### PR TITLE
[7.4-only] LPS-125256 Remove deprecated tag attributes

### PR DESCRIPTION
**Pull request by julien**
Deprecated attributes have been removed from the following tags and
there usages:

- `<clay:alert />`
- `<clay:badge />`
- `<clay:button />`
- `<clay:checkbox />`
- `<clay:dropdown-actions />`
- `<clay:dropdown-menu />`
- `<clay:icon />`
- `<clay:label />`
- `<clay:link />`
- `<clay:navigation-bar />`
- `<clay:progressbar />`
- `<clay:radio />`
- `<clay:sticker />`
- `<clay:stripe />`

/cc @liferay-lima, @liferay-echo, @liferay-commerce

### CODE REVIEW

**Comment by julien** [comment]
ci:stop

**Comment by julien** [comment]
ci:stop

**Comment by julien** [comment]
When I run `ant format-source-current-branch` from the [portal-impl](https://github.com/liferay/liferay-portal/tree/cbf9817ba745154f6a108a14f0d16c359461fb2c/portal-impl) directory locally, I get 3 changes and that's why `ci:test:sf` was failing, my bad.

But I also get another error from [a file](https://github.com/liferay/liferay-portal/blob/cbf9817ba745154f6a108a14f0d16c359461fb2c/modules/apps/frontend-js/frontend-js-loader-modules-extender/src/main/java/com/liferay/frontend/js/loader/modules/extender/internal/servlet/taglib/JSLoaderTopHeadDynamicInclude.java) I didn't touch in this PR.

```
$ ant format-source-current-branch
Buildfile: /home/jc/Documents/portal/liferay-portal/portal-impl/build.xml
   [delete] Deleting: /home/jc/Documents/portal/liferay-portal/null796571390.properties

format-source-current-branch:
   [delete] Deleting: /home/jc/Documents/portal/liferay-portal/null796571390.properties

format-source-files:
     [java] Loading file:/home/jc/Documents/portal/liferay-portal/portal-impl/classes/system.properties
     [java] SLF4J: Class path contains multiple SLF4J bindings.
     [java] SLF4J: Found binding in [file:/home/jc/Documents/portal/liferay-portal/util-slf4j/classes/org/slf4j/impl/StaticLoggerBinder.class]
     [java] SLF4J: Found binding in [jar:file:/home/jc/Documents/portal/liferay-portal/lib/development/slf4j-simple.jar!/org/slf4j/impl/StaticLoggerBinder.class]
     [java] SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
     [java] SLF4J: Actual binding is of type [com.liferay.util.slf4j.LiferayLoggerFactory]
     [java] There should be an empty line before line '66', as we finished referencing variable 'printWriter', see https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/checks/missing_empty_line_check.markdown: ./modules/apps/frontend-js/frontend-js-loader-modules-extender/src/main/java/com/liferay/frontend/js/loader/modules/extender/internal/servlet/taglib/JSLoaderTopHeadDynamicInclude.java 66 (Checkstyle:MissingEmptyLineCheck)
     [java] java.lang.Exception: Found 1 formatting issues:
     [java] 1: There should be an empty line before line '66', as we finished referencing variable 'printWriter', see https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/checks/missing_empty_line_check.markdown: ./modules/apps/frontend-js/frontend-js-loader-modules-extender/src/main/java/com/liferay/frontend/js/loader/modules/extender/internal/servlet/taglib/JSLoaderTopHeadDynamicInclude.java 66 (Checkstyle:MissingEmptyLineCheck)
     [java]
     [java]     at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:463)
     [java]     at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:276)

BUILD FAILED
/home/jc/Documents/portal/liferay-portal/portal-impl/build.xml:572: The following error occurred while executing this line:
/home/jc/Documents/portal/liferay-portal/portal-impl/build.xml:706: Java returned: 1

Total time: 1 minute 16 seconds
```

There's a missing empty line in my [LPS-125256](https://github.com/julien/liferay-portal/blob/LPS-125256/modules/apps/frontend-js/frontend-js-loader-modules-extender/src/main/java/com/liferay/frontend/js/loader/modules/extender/internal/servlet/taglib/JSLoaderTopHeadDynamicInclude.java#L66) branch, which hasn't been rebased with [upstream](https://github.com/liferay/liferay-portal/blob/cbf9817ba745154f6a108a14f0d16c359461fb2c/modules/apps/frontend-js/frontend-js-loader-modules-extender/src/main/java/com/liferay/frontend/js/loader/modules/extender/internal/servlet/taglib/JSLoaderTopHeadDynamicInclude.java#L67), so I'll do that now.

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/info_panel_file_entry.jsp [hunk 1]
```diff
@@ -196,7 +196,6 @@ long assetClassPK = DLAssetHelperUtil.getAssetClassPK(fileEntry, fileVersion);
 													href="<%= DLURLHelperUtil.getDownloadURL(fileEntry, fileVersion, themeDisplay, StringPool.BLANK, false, true) %>"
 													label="download"
 													small="<%= true %>"
-													title='<%= LanguageUtil.format(resourceBundle, "file-size-x", LanguageUtil.formatStorageSize(fileVersion.getSize(), locale), false) %>'
```
Comment (julien) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/info_panel_file_entry.jsp:199]: Note to self: revert this change (whole file)

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/search_info.jsp [hunk 2]
```diff
@@ -82,7 +82,6 @@ List<Folder> mountFolders = DLAppServiceUtil.getMountFolders(scopeGroupId, DLFol
 					href="<%= searchEverywhereURL.toString() %>"
 					label="everywhere"
 					small="<%= true %>"
-					title="everywhere"
```
Comment (julien) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/search_info.jsp:85]: Revert

File: modules/apps/fragment/fragment-web/src/main/resources/META-INF/resources/js/actions.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
```
Comment (julien) [inline modules/apps/fragment/fragment-web/src/main/resources/META-INF/resources/js/actions.js:16]: 2 license headers?

File: modules/apps/comment/comment-taglib/src/main/resources/META-INF/resources/discussion/view_message_thread.jsp [hunk 4]
```diff
@@ -122,7 +122,6 @@ Format dateFormatDateTime = FastDateFormatFactoryUtil.getDateTime(locale, timeZo
 									</liferay-util:buffer>
 
 									<clay:link
-										ariaLabel='<%= LanguageUtil.format(request, "in-reply-to-x", HtmlUtil.escape(parentDiscussionComment.getUserName()), false) %>'
```
Comment (julien) [inline modules/apps/comment/comment-taglib/src/main/resources/META-INF/resources/discussion/view_message_thread.jsp:125]: Change to `aria-label` don't remove

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/dropdowns.jsp [hunk 5]
```diff
@@ -47,7 +47,6 @@
 		<clay:dropdown-menu
```
Comment (julien) [inline modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/dropdowns.jsp:47]: This file is important for QA tests

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/links.jsp [hunk 6]
```diff
@@ -30,14 +30,12 @@
...
 	label="link text"
-	target="blank"
 />
 
 <clay:link
 	displayType="primary"
 	href="#"
 	label="a button link"
 	rel="nofollow"
-	target="blank"
```
Comment (julien) [inline modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/links.jsp:41]: Can probably leave these, they should be passed as dynamic attributes, check

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/AlertTag.java [hunk 7]
```diff
@@ -125,27 +79,10 @@ public void setMessage(String message) {
...
 	}
 
-	public void setType(String type) {
-		_type = type;
-	}
-
-	/**
-	 * @deprecated As of Athanasius (7.3.x), with no direct replacement
-	 */
-	@Deprecated
```
Comment (julien) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/AlertTag.java:148]: Remember to comment that this is because it's used by `StripeTag` so we un-deprecate

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java [hunk 8]
```diff
@@ -38,33 +37,10 @@ public int doStartTag() throws JspException {
...
 
-		if (Validator.isNotNull(_ariaLabel)) {
-			setDynamicAttribute(StringPool.BLANK, "aria-label", _ariaLabel);
-		}
-
 		if (Validator.isNotNull(_href)) {
 			setDynamicAttribute(StringPool.BLANK, "href", _href);
 		}
 
-		if (Validator.isNotNull(_target)) {
```
Comment (julien) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java:49]: Check the `target` from dynamic attributes, review this whole method, can't remove everything

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/liferay-clay.tld [hunk 9]
```diff
@@ -298,21 +154,15 @@
...
 		<attribute>
 			<description></description>
 			<name>propsTransformer</name>
 			<required>false</required>
 			<rtexprvalue>true</rtexprvalue>
 		</attribute>
 		<attribute>
-			<description>Deprecated as of 7.3.0, replaced by the attribute <![CDATA[<code>small</code>]]>.</description>
-			<name>size</name>
+			<description></description>
```
Comment (julien) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/liferay-clay.tld:164]: This looks broken

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by wincent** [comment]
I only skimmed this because it's still a draft (and I am a taglib noob), but just wanted to add a reminder that a new entry in [`readme/BREAKING_CHANGES.markdown`](https://github.com/liferay/liferay-portal/blob/master/readme/BREAKING_CHANGES.markdown) should probably go in with this too.

**Comment by julien** [comment]
Thanks @wincent I had totally forgotten about that.

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by julien** [comment]
ci:test:bundle

**Comment by julien** [comment]
Hi @john-co and @PablitoBonito,

I'm looking at the unique failures for this PR and wanted to tell you what I saw.
I ran all the tests locally (or at least tried to) with the following command:

```sh
ant -f build-test.xml run-selenium-test -Dtest.class=TEST_NAME
```

For example:

```sh
ant -f build-test.xml run-selenium-test -Dtest.class=LocalFile.DepotRemoteStagingWC#EditImageInDepot
```

Here's a summary of what I saw for each "unique" failure:

**LocalFile.DepotRemoteStagingWC#EditImageInDepot**

Tries to connect to http://localhost:9080, don't know if that's expected

LocalFile DepotRemoteStagingWC#EditImageInDepot-02

Trying to connect an asset library to a site works "manually":

LocalFile DepotRemoteStagingWC#EditImageInDepot

I'll look for `modalIframeLoaded` in the changes because of this error message:

	`Event "modalIframeLoaded" with attribute "src" and value "dialog-iframe-popup" not fired.`

**LocalFile.DepotLocalStagingDM#DeleteImageInDepot**
Looks more like a warning that an actual error

**LocalFile.DepotDisplayPage#ViewWC**
Not sure what's happening

**LocalFile.DepotIntegration#Smoke**
Seems similar to **LocalFile.DepotRemoteStagingWC#EditImageInDepot**

**LocalFile.DepotDisplayPage#ViewWC**
	```
	Cause:
	LIFERAY_ERROR: java.lang.RuntimeException: org.elasticsearch.ElasticsearchStatusException: 
	[liferay-52178-workflow-metrics-transitions/qjWqlaJsSlKvRqvLSc6MnA] 
	ElasticsearchStatusException[Elasticsearch exception 
	[type=resource_already_exists_exception, reason=index [liferay-52178-workflow-metrics-transitions/qjWqlaJsSlKvRqvLSc6MnA] already exists]]
	```
Is that an error in the test itself?

**LocalFile.DepotMemberships#AddAndRemoveUser**

Here a some screenshots when testing manually 

LocalFile DepotMemberships#AddAndRemoveUser-01

LocalFile DepotMemberships#AddAndRemoveUser-02

**LocalFile.DepotLocalStagingDM#EditDocumentInDepot**
Seems similar to **LocalFile.DepotDisplayPage#ViewWC**

**LocalFile.DepotWorkflow#ApproveImage**
Seems similar to **LocalFile.DepotRemoteStagingWC#EditImageInDepot**

**LocalFile.DepotWorkflow#Smoke**
Seems similar to **LocalFile.DepotRemoteStagingWC#EditImageInDepot**

**LocalFile.DepotWorkflow#ApproveWCArticle**
Seems similar to **LocalFile.DepotRemoteStagingWC#EditImageInDepot**

**LocalFile.Fragments#ViewCannotDeleteFragmentInUse**
	```
	Cause:
	Element is not present at "//*[contains(@class,'btn')][normalize-space(text())='New']"
	```
Is this because of a markup change?

I just generated a bundle with `ci:test:bundle` just in case I can be able to see more.
Thanks

**Comment by julien** [comment]
ci:reevaluate:1132004_5314

**Comment by julien** [comment]
@john-co, @PablitoBonito just so you know, I found the cause for the **LocalFile.DepotRemoteStagingWC#EditImageInDepot** test so I will send a fix.

**Comment by julien** [comment]
@john-co @PablitoBonito I had another look and the only test I'm not able to fix is **LocalFile.Fragments#ViewCannotDeleteFragmentInUse**: the markup has been changed.

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by julien** [comment]
Just realized I forgot to add a file.

**Comment by julien** [comment]
ci:stop

**Comment by julien** [comment]
ci:stop

**Comment by julien** [comment]
ci:stop

**Comment by julien** [comment]
ci:test:relevant

**Comment by john-co** [comment]
ci:test:bundle

**Comment by john-co** [comment]
Hi Julien,

[Related]
- Fragments#ViewCannotDeleteFragmentInUse: looks like the markup as you said has changed. I was able to reproduce the issue locally. Here are some differences I was able to find. I'm not exactly sure where in the code it changed from btn to link-primary class here but hopefully this helps.

screenshot_2
image

**Comment by julien** [comment]
Hey @john-co thanks a lot; that does help a lot because some of those errors are exactly the ones I checked and I thought I had fixed.

I'll compile (that branch) here locally and get back to you if I have any doubts.

**Comment by julien** [comment]
> Fragments#ViewCannotDeleteFragmentInUse: looks like the markup as you said has changed. I was able to reproduce the issue locally. Here are some differences I was able to find. I'm not exactly sure where in the code it changed from btn to link-primary class here but hopefully this helps.

That one should be fixed, with [this latest commit](https://github.com/liferay-frontend/liferay-portal/pull/716/commits/f37c2f982a25cc6f3837d7ac050521c2f20ffbb3)



> DepotMemberships#AddAndRemoveUser - Clicking on Remove Membership dropdown does not do anything.

That's "funny", because I'm pretty sure it wasn't touched in this PR but was done when we updated the cards (from v2 to v3) (as part of [this commit](https://github.com/liferay/liferay-portal/commit/71431d935940a0387956c53cb0457ab3dd97ff8e)) : so if you switch to "table" or "list" view, you should see it working, but when in "card" view it indeed fails silently. Regardless, I've fixed that and it should be visible in this gif



Anyway, I guess the idea is "don't talk, show me the test results" so I'll re-run ci tests.
Additionally and just so everyone knows, I'll squash these "fix" commits into the first one before forwrarding.

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
Almost there @julien!

**Comment by julien** [comment]
Let's see if I'm able to fix those last failures :muscle:

**Comment by julien** [comment]
@john-co, @PablitoBonito for 

**LocalFile.MastersPageTemplates#ChangeMasterOfContentPageAmongExistingMasters**

This seems more like a warning than an error, I don't think my changes affects that functionality and I was able to change the master page for a page I created



**LocalFile.MastersPageTemplates#ChangeMasterOfContentPageTemplateAmongExistingMasters**

This also looks like a warning, but I'll need your help to confirm



**LocalFile.MastersPageTemplates#ChangeMasterOfDisplayPageTemplateAmongExistingMasters**
and **LocalFile.MastersPageTemplates#EditMasterPageTemplateViaPageEditorLookAndFeel** : 

I think this is also unrelated, I have a feeling that the markup might have changed and that those tests might need to be updated, but not 100% sure. 

Note that I haven't rebased with master today so that might be it.

**Comment by john-co** [comment]
ci:test:bundle

**Comment by john-co** [comment]
Just started reviewing :)

**Comment by john-co** [comment]
[Related]
- MasterPages#* - clicking on Change Master button opens the iframe but it immediately closes. js console error is thrown
```
DevTools failed to load SourceMap: Could not load content for http://localhost:8080/main.css.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```
ChangeMasterIframe

**Comment by julien** [comment]
Thanks @john-co there are different ways the master page can be changed and I wasn't looking in the correct place. I also updated the code for other buttons, allowing us to remove 2 `<aui:script>` tags in the JSP

With my last change it should be fixed, but I rebased with master just in case.

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by john-co** [comment]
ci:test:bundle

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by julien** [comment]
ci:test:bundle

**Comment by julien** [comment]
@john-co when you can, I'm going to need your help (again) on to analyse the errors. thanks in advance.

**Comment by julien** [comment]
ci:test:bundle

**Comment by john-co** [comment]
Just started reviewing :)

**Comment by john-co** [comment]
Hi @julien, here's what I could find out.

[Related]
- MastersPageTemplates#* - looks like the test is expecting the Themes section to be present immediately when the master name is changed back to Blank. I noticed that the Theme section disappears immediately after master test template is selected (seems expected?). But the Theme section does not appear when Blank is selected, only until the Look and Feel config is saved will the Theme section come back.
ChangeMasterTheme

Here's what we're expecting:
ExpectedMasterTheme

I also noticed that the primary button color is missing from the Change Master and Change Style Book buttons.
screenshot_4

**Comment by julien** [comment]
@john-co thanks a lot (again) for the help, that **MastersPageTemplates#*** issue should be fixed (and the styles for both buttons too)

recording

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by julien** [comment]
ci:test:bundle

**Comment by julien** [comment]
Looking at the poshi reports I see this 



which doesn't seem to be related with my changes.

**Comment by julien** [comment]
Trying to access https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=21,label_exp=!master/215552//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_DepotRemoteStagingDM_DeleteImageInDepot_ but it's a 404

**Comment by john-co** [comment]
Just started reviewing :)

**Comment by john-co** [comment]
[Related]
- MastersPageTemplates#ChangeMasterOfWidgetPageTemplateAmongExistingMasters - for some reason there's that permission access issue when trying to configure a Widget Page Template. This issue doesn't occur with a Content Page Template. I'm not able to reproduce the issue on master branch and I see that the test has failed [on this PR in the past 3 runs](https://testray.liferay.com/home/-/testray/cases/1310783519). So I'm inclined to say that this issue is related to the PR somehow.

WidgetPageTemplatePermissions

**Comment by julien** [comment]
Hi @john-co thanks again

That issue should be fixed

recording

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by julien** [comment]
ci:test:bundle

**Comment by john-co** [comment]
ci:stop

**Comment by john-co** [comment]
Hi @julien , looks like `ci:test:relevant` got stuck. Before we rerun the test, could you also rebase to fix the merge conflict?

**Comment by john-co** [comment]
I asked ci channel about the test suite hanging in https://liferay.slack.com/archives/C19PWQ74J/p1612369644034700. Seems like this is not the only PR affected by tests stuck on ci.

**Comment by julien** [comment]
@john-co sure I'll fix the conflict, thanks for warning :+1:

**Comment by julien** [comment]
ci:test:relevant

**Comment by julien** [comment]
ci:test:relevant

**Comment by julien** [comment]
Hey @john-co looking at the test failures, I'm not sure about these.

1. **test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/2**
  Doesn't seem related but I may be wrong

2. **test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/1**
 
     **LocalFile.DMAutoTagging#Smoke - Poshi Report - Poshi Summary - Console Output**
      This might be but I'm not sure, I'll have a look


3. **test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8/0**
   Doesn't seem related

**Comment by julien** [comment]
ci:test:bundle

**Comment by john-co** [comment]
Hi @julien,

I took a look and the unique failures don't seem related to the PR. The DMAutoTagging test seems flaky from [test history](https://testray.liferay.com/home/-/testray/cases/1310734985) . May manually forward if ready.

Reviewed. Test failures are unrelated. :heavy_check_mark:

**Comment by john-co** [comment]
search test fails intermittently: https://issues.liferay.com/browse/LPS-123052

**Comment by john-co** [comment]
cc/ @xbrianlee maybe some way to tag flaky integration/unit tests would be good to ignore from ci:test:relevant results. Our team runs into flaky integration test batch involving search once in a while.

**Comment by john-co** [comment]
ci:test:relevant

**Comment by julien** [comment]
Hi @john-co ,

That's good news, thanks again for all the help.

Let's see if the rest of the team has something to say and maybe this can be forwarded.

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/view_asset_list_items.jsp [hunk 10]
```diff
@@ -100,11 +100,6 @@ ListItemsActionDropdownItems listItemsActionDropdownItems = (ListItemsActionDrop
 	</liferay-ui:search-container>
 </clay:container-fluid>
 
-<liferay-frontend:component
```
Comment (jbalsas) [inline modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/view_asset_list_items.jsp:103]: This is nice!

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/header/page.jsp [hunk 11]
```diff
@@ -159,11 +159,11 @@ String myWorkflowTasksPortletNamespace = PortalUtil.getPortletNamespace(PortletK
 						<div class="dropdown-menu dropdown-menu-right" id="<portlet:namespace />commerce-dropdown-assigned-to">
 							<c:if test="<%= !assignedToCurrentUser %>">
 								<clay:button
-									elementClasses="dropdown-item transition-link"
+									cssClass="dropdown-item transition-link"
+									displayType="secondary"
 									id='<%= liferayPortletResponse.getNamespace() + "assign-to-me-modal-opener" %>'
 									label='<%= LanguageUtil.get(request, "assign-to-me") %>'
```
Comment (jbalsas) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/header/page.jsp:165]: Do we need `LanguageUtil` in `label`? I think the new tags support automatic translation, so just `label="assign-to-me"` should do the work and make it easier?

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/header/page.jsp [hunk 12]
```diff
@@ -328,7 +328,7 @@ String myWorkflowTasksPortletNamespace = PortalUtil.getPortletNamespace(PortletK
 
 					<c:if test="<%= Validator.isNotNull(previewUrl) %>">
 						<clay:link
-							elementClasses="btn btn-outline-borderless btn-outline-secondary btn-sm text-primary"
+							cssClass="btn btn-outline-borderless btn-outline-secondary btn-sm text-primary"
```
Comment (jbalsas) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/header/page.jsp:331]: I think many of these can be accomplished using the API with `type="button"` and `displayType`, `borderless` and `size` rather than custom css classes.

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/panel/start.jsp [hunk 13]
```diff
@@ -50,7 +50,7 @@ String collapseSwitchId = Validator.isNotNull(collapseSwitchName) ? collapseSwit
 				</c:when>
 				<c:when test="<%= Validator.isNotNull(actionIcon) %>">
 					<clay:link
-						elementClasses="btn btn-monospaced btn-primary btn-sm text-white"
+						cssClass="btn btn-monospaced btn-primary btn-sm text-white"
```
Comment (jbalsas) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/panel/start.jsp:53]: Same as the other `<clay:link` case and probably others

File: modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/pending_orders/b2b/view.jsp [hunk 14]
```diff
@@ -55,11 +55,11 @@ CommerceAccount commerceAccount = commerceOrderContentDisplayContext.getCommerce
 				<aui:input name="deleteCommerceOrderIds" type="hidden" />
 
 				<clay:button
+					cssClass="btn-fixed btn-primary"
```
Comment (jbalsas) [inline modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/pending_orders/b2b/view.jsp:58]: `btn-primary` should be added with `displayType` and not as a custom class

**Comment by jbalsas** [comment]
Hey @julien, nice job on this one!!

I've only skimmed it since it is massive, but it overall looks good to me!!

I'm going to forward this and then I'd suggest we follow up once it's merged reviewing some of the things I just pointed out to consolidate usages and avoid bad examples out there that get copied over to other places.

**Comment by julien** [comment]
@jbalsas thanks a lot for that review. 

As you said this was massive, so lesson learned on my side: split changes per module or in this case per tag, and a second point would be to take a bit more time to think using the correct attributes.

As soon as it's possible I'll apply the request changes.

**Comment by jbalsas** [comment]
ci:forward
